hardware/raspberrypi/bootmodes/msd.md

```diff
@@ -4,7 +4,20 @@
 
 This tutorial explains how to boot your Raspberry Pi from a USB mass storage device such as a flash drive or a USB hard disk. Note that this feature does not work with all USB mass storage devices.
 
-## Program USB host boot mode (Raspberry Pi 3A+ and 3B only)
+See the [documentation](https://www.raspberrypi.org/documentation/hardware/raspberrypi/bootmodes/) for the boot sequence and alternative boot modes (Network, USB device, GPIO or SD boot).
+
+Note that "USB Mass Storage Boot" is different from "USB Device Boot Mode", the device boot mode allows a Raspberry Pi connected to a computer to boot using files on that computer.
+
+## Raspberry Pi 1, 2 (V1.1), Compute Module, Zero
+
+**These models are not supported for USB Mass Storage Boot.**
+The boot code for USB is stored in the BCM2837 device only, so the Pi 1 A/B, Pi 2 B (v1.1), and Pi Zero will all require SD cards as they are based on the BCM2835 and BCM2836. This boot code is stored in ROM (except Pi 4B) which by definition cannot be changed.
```

Comment:
Also the Pi1 A+/B+, the Pi Zero W and the CM1. Rather than listing the models that **don't** support MSD boot without an SD card, it might be easier to list only those models that **do**? (and then say something like "all other models of Raspberry Pi will need to use the 'special bootcode.bin-only boot mode'") :man_shrugging:

Reply:
FIrst paragraph - not needed, since again this page will usually be found via the boot modes page, which includes a link to device mode boot which explains what device mode boot does.

The rest is not needed and will only add confusion - the first paragraph of the page already specifies the models which the page applies to.

Reply:
for the rpi4, the boot code is stored on both rom and spi flash

the rpi4 rom appears to only be able to run code from 2 places: `recovery.bin` on an SD card, or a blob in the SPI flash

and then that SPI flash would add the usb support (i think its currently missing)

Reply:
Cleverca22

Yes; it can boot from SPI EEPROM (I think SPI Flash is incorrect as that more describes an SD card). But it currently does not support USB boot. I've added a link to the Pi 4B EEPROM boot page; to not have it documented twice and have one place with the complete description.

See: https://www.raspberrypi.org/documentation/hardware/raspberrypi/booteeprom.md